src/main/java/com/unstoppabledomains/resolution/NamingService.java

```diff
@@ -3,13 +3,13 @@
 import com.unstoppabledomains.exceptions.NamingServiceException;
 
 abstract class NamingService {
-    abstract Boolean isSupported(String domain);
-    abstract String addr(String domain, String ticker) throws NamingServiceException;
-    abstract String ipfsHash(String domain) throws NamingServiceException;
-    abstract String email(String domain) throws NamingServiceException;
-    abstract String owner(String domain) throws NamingServiceException;
+    protected abstract Boolean isSupported(String domain);
+    protected abstract String addr(String domain, String ticker) throws NamingServiceException;
+    protected abstract String ipfsHash(String domain) throws NamingServiceException;
+    protected abstract String email(String domain) throws NamingServiceException;
+    protected abstract String owner(String domain) throws NamingServiceException;
 
-    public String namehash(String domain) {
+    protected String namehash(String domain) {
         return NameHash.nameHash(domain);
```

Comment:
Let's keep casing consistent. We usually write `namehash` without capitalizing the "h"